### ISSUE TITLE
Suggestion: Add historical record of important towny server events

### ISSUE THREAD

**Issue by Goosius1** [TownyAdvanced/Towny#3555]
**DESCRIPTION**
- Something I often look for on a server is a 'historical record' of important events.
- When I join a new server, I might  like to know how active it is, and even a small list of things like "towns founded recently", would helps improve my first impression.
- Also I might like to know some details of specific towns and nations:
1. How has a nation changed over time? e.g. how quickly has the nation grown?, what did the succession of kings look like?, has it been a peaceful or warlike nation? 
2. How has a town changed over time? e.g. how quickly has the town grown?, has the town treasury always been stable?, what did the succession of mayors look like?, how often has the town been attacked by enemies.
- Also, on a more aesthetic note, I would like to know where the constructed 'wonders' of the server are, and when were they built  (& possibly a brief description of what they mean).

**USE CASES**
- I see that Earth MC already manually maintains such a historical list continuously, which I find is great to read for seeing what is happening.  So at a minimum, this suggested feature could help reduce that manual effort:  https://earthmc.fandom.com/wiki/Flooding_Era
- On Minesuperior (& likely other servers), the server is filled with amazing structures, but many players have no idea that they exist. It seems likely to improve user experience to provide an easy-to access list.
- On Minesuperior (& likely other servers),  there are many huge ruined towns where the history has been effectively completely lost. For me anyway, I think it would be great to arrive at a ruin and get a history of the area.

**SOLUTION**
- These records would have to be persisted in a data store somewhere.
- The storage design would be important, because we would want players to be able to access the information without performance loss to the DB.
- Certain vectors of data retrieval would be more important than others e.g.  it could be a popular search to get the history of everything related to 1 particular Town.
- The idea of recording 'wonders' requires some extra innovation. How would certain structures be tagged as wonders?.  Perhaps server mods/admins would do the tagging?   Or perhaps it would be down to votes by the playerbase?

**Comment by LlmDl** [comment]
When I ran my server we maintained a library in the spawn city to handle this. Before MC had writable books there was a plugin called Bookworm which allowed players to write, copy and store books in Bookcase blocks.

This could be a feature in Towny but, as it is mainly role-playing, and whatever data we did record would end up being a much drier/more boring read than something written by a real human being, I don't know if this should be something we add. RP features usually end up being fairly difficult to bring into a concrete feature set that can be added while making all the RPers happy.

I'll leave this up for community feedback.
